More checkpoint related fixes

vgeorge opened the pull request:
Contributes to:

- #120

Also addresses:

- On checkpoint switch, only apply timeframe if present
- Hide delete button when there is no mosaic to delete
- Fix timeframe issue when deleting the last timeframe for a checkpoint (now it will display empty state or apply an existing one if checkpoint id match)

How to test: On an existing project, create multiple checkpoints and timeframes for each of them. Try switching between them and check if the latest timeframe is from the selected checkpoint. Then delete all timeframes from a checkpoint and check if the proper message is displayed. Check if switching checkpoints still works.

@LanesGood ready for review.

LanesGood reviewed (commented):
Looking really good. I think I was able to fulfill all of the test scenarios you described, but I hit one issue:

After saving a new checkpoint, I'm able to add predictions on additional timeframes in my saved checkpoint. However, when switching back to the base model, my checkpoint (saved and previously visible in the checkpoint list) disappears. After reloading I see my saved checkpoint again. Can/should this be addressed here?

vgeorge commented:
@LanesGood I added a fix for the issue you found. I also added the following changes:

- When drawing a new AOI on an existing project, hide the timeframe list during drawing. The list is not deleted from the state because AOI draw can be canceled
- Switching between AOIs refreshes the timeframe list properly

To test: on a project with many timeframes, start drawing a new AOI, the timeframe list should be empty during drawing. Draw a really small AOI to cancel drawing, the former selected AOI should be reinstated. Switch between existing AOIs, it should select update the timeframes list correctly.

This is ready for another review. I believe there are some issues with Batch AOIs, but I'll address them in separate.

LanesGood reviewed (approved):
@vgeorge nice - this does now seem to fix many of the challenging states with switching between AOI's, checkpoints and timeframes. Thanks for untangling the yarn here.
